src/linker/pipeline_schema.py

```diff
@@ -26,19 +28,14 @@ def get_schemas(cls) -> List["PipelineSchema"]:
     def _add_step(self, step: Step) -> None:
         self.steps.append(step)
 
-    def validate_pipeline(self, pipeline: Pipeline) -> bool:
-        # Loop through the pipeline implementations and check if their order
-        ## and corresponding steps match the schema
-        for idx, implementation in enumerate(pipeline.implementations):
-            if implementation.step != self.steps[idx]:
-                return False
-        return True
+
+_PIPELINE_SCHEMAS = PipelineSchema.get_schemas()
```

Comment:
If we expose this, we can have `pipeline.validate()` refer to `PIPELINE_SCHEMAS` naturally.